opentracing/src/main/java/opentracing/TraceContextDecoder.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Decodes a trace context from binary or text formats.
+ *
+ * @see TraceContextEncoder
+ */
+// See https://github.com/opentracing/opentracing.github.io/issues/21 about folding codec together
+public interface TraceContextDecoder {
```

Comment:
> re supporting text and binary but nothing else: TBH, I would probably
have just gone for binary and told folks that wanted to stuff them in HTTP
headers to base64 encode or whatever. But my understanding is that many
zipkin users really like having the various fields split out in
human-readable (and software-router-accessible) fashion in HTTP headers,
hence the text format. We hope (?) that there will only be two, hence
cramming them into one interface.
I dont know how many are processing zipkin headers as intermediaries, but
there's certainly a support use case of taking the value of the trace id
header and plopping it into a search box.